src/ra_server.erl

```diff
@@ -919,6 +919,20 @@ handle_candidate(#pre_vote_rpc{term = Term} = Msg,
 handle_candidate(#request_vote_rpc{}, State = #{current_term := Term}) ->
     Reply = #request_vote_result{term = Term, vote_granted = false},
     {candidate, State, [{reply, Reply}]};
+handle_candidate(#pre_vote_rpc{term = Term, last_log_index = PeerLastIdx} = PreVote,
+                 #{current_term := CurTerm,
+                   cfg := #cfg{log_id = LogId}} = State)
+  when Term =:= CurTerm ->
+    %% This clause is necessary to prevent the liveness issue reported in https://github.com/rabbitmq/ra/issues/439
+    {LastIdx, _} = last_idx_term(State),
+    case PeerLastIdx > LastIdx of
```

Comment:
No. I added the check just because keeping the current behavior (i.e., `candidate` ignores `pre_vote_rpc` not having greater term) as much as possible. If you think it is not necessary, it is better to remove the extra check.

Reply:
Fixed in commit https://github.com/rabbitmq/ra/pull/442/commits/840f7ba48221f20c101524c8f037d02155d8b312.